### ISSUE TITLE
By-Election results

### ISSUE THREAD

**Issue by jrstarke** [j3camero/canada-election-forecast#1]
There have been a number of By-Elections since 2011, so in a number of areas, the 2011 results no longer represent the most up to date information about the voting in that region. 

https://en.wikipedia.org/wiki/By-elections_to_the_41st_Canadian_Parliament

For example, in Trinity-Spadina, in the 2011 election, it was Highly NDP, Olivia Chow winning by a margin of 31.12%. When she stepped down in 2014 to run for Toronto Mayor, the by-election was won by a margin of 19.52%.

https://en.wikipedia.org/wiki/By-elections_to_the_41st_Canadian_Parliament#Trinity.E2.80.94Spadina

As this region was redistricted for the latest boundaries, this would likely effect the proportions in _Spadina--Fort York_ and _University--Rosedale_ ridings, as the Liberals have both done better in that region more recently, and the Liberals have done considerably better in the polls.

**Comment by jrstarke** [comment]
Of course, looking at the actual numbers, there is little reason to vote strategically, as both of these parties individually do better than the Conservatives.

**Comment by jrstarke** [comment]
Based on the 2012 By-Election, my previous point about there being little reason to vote strategically in the Toronto ridings, would also likely apply in _Victoria_ as well, since the NDP won over the Green by a 2.87% margin, with the Conservatives being 19.81% behind.

https://en.wikipedia.org/wiki/By-elections_to_the_41st_Canadian_Parliament#Victoria

**Comment by j3camero** [comment]
Thanks so much for this by-election info. It's truly one of the biggest
fuck-ups that the model has made so far, and I'm coding furiously right now
to fix it.

**Comment by j3camero** [comment]
Progress update. The changes made to fix another issue (https://github.com/j3camero/canada-election-forecast/issues/5) have had an effect on this one. Many of the ridings have by-election data coming in along with riding-specific poll data. This issue is now fixed for many ridings, but not all.

My plan to fix the issue for all ridings will be to add a file with the by-election results that gets pulled in by the model and treated similarly to recent riding-specific polls.

**Comment by KelseyDH** [comment]
Yeah just to note I've had multiple people attack the accuracy of the site for the predictions the site is making about Victoria.  Even if it's not an ABC race, it's affecting perception of the site's reliability.
